modules/tinymce/src/plugins/media/test/ts/browser/NoneditableRootTest.ts

```diff
@@ -1,39 +1,256 @@
+      TinyUiActions.submitDialog(editor);
+
+      await Waiter.pTryUntil('Wait for structure check', () => {
+        TinyAssertions.assertContentStructure(editor, ApproxStructure.build((s, str, arr) => {
+          return s.element('div', {
+            children: [
+              s.element('div', {
+                classes: [ arr.has('mceEditable') ],
+                attrs: {
+                  contenteditable: str.is('true')
+                },
+                children: [
+                  s.element('div', {
+                    children: [
+                      s.element('h1', {
+                        children: [
+                          s.text(str.is('Test'))
+                        ]
+                      }),
+                      s.element('div', {
+                        children: [
+                          s.element('span', {
+                            classes: [ arr.has('mce-preview-object'), arr.has('mce-object-iframe') ],
+                            attrs: {
+                              'contenteditable': str.is('false'),
+                              'data-mce-object': str.is('iframe'),
+                              'data-mce-selected': str.is('1')
+                            },
+                            children: [
+                              s.element('iframe', {
+                                attrs: {
+                                  src: str.is('https://www.youtube.com/embed/b3XFjWInBog')
+                                }
+                              }),
+                              s.element('span', {})
+                            ]
+                          })
+                        ]
+                      }),
+                    ],
+                  })
+                ]
+              }),
+              s.theRest()
+            ]
+          });
+        }));
+      });
+      TinyAssertions.assertSelection(editor, [ 0, 0, 1 ], 0, [ 0, 0, 1 ], 1);
+    });
+  });
+
+  context('Inline mode editable_root', () => {
+    const hook = TinyHooks.bddSetup<Editor>({
+      plugins: 'media',
+      inline: true,
+      toolbar: 'media',
+      contextmenu: 'media',
+      base_url: '/project/tinymce/js/tinymce'
+    }, [ Plugin ], true);
+
+    it('TINY-10820: Saving media dialog should not cause duplicate media when using editable_root: true', async () => {
+      const editor = hook.editor();
+      editor.focus();
+      const media = `<iframe src="https://www.youtube.com/embed/b3XFjWInBog" width="560" height="314" frameborder="0" allowfullscreen="allowfullscreen"></iframe>`;
+      editor.setContent('<div class="mceEditable" contenteditable="true"><div class="random"><h1 style="text-align: center;">Test</h1><div id="test">' + media + '</div></div></div>');
+      const targetElement = UiFinder.findIn<HTMLElement>(TinyDom.body(editor), '.mce-preview-object').getOrDie();
+      targetElement.dom.focus();
+      Mouse.trueClick(targetElement);
+      Mouse.trueClick(targetElement);
```

Comment:
We do we need to click twice? Could we adda comment on that if it's needed.

Reply:
Clicking once will still do the `focusBody`, which will send the selection to the start of the editable block since there was no bookmark once the editor loaded. So doing another click will only actually sending focusing on the contenteditable block.